### ISSUE TITLE
datetime.strptime %Z doesn't produce aware object

### ISSUE THREAD

**Issue by 9505c713-bcdd-4f76-9351-74d1a4b0dea9** [python/cpython#72751]
BPO | [28565](https://bugs.python.org/issue28565)
--- | :---
Nosy | @abalkin, @lilydjwg

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2016-10-31.05:48:57.512>
labels = ['3.7', 'type-feature', 'library']
title = "datetime.strptime %Z doesn't produce aware object"
updated_at = <Date 2016-10-31.09:43:41.614>
user = 'https://github.com/lilydjwg'
```

bugs.python.org fields:
```python
activity = <Date 2016-10-31.09:43:41.614>
actor = 'SilentGhost'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2016-10-31.05:48:57.512>
creator = 'lilydjwg'
dependencies = []
files = []
hgrepos = []
issue_num = 28565
keywords = []
message_count = 2.0
messages = ['279761', '279775']
nosy_count = 3.0
nosy_names = ['belopolsky', 'SilentGhost', 'lilydjwg']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue28565'
versions = ['Python 3.7']
```

</p></details>

**Comment by pganssle** [comment]
A lot of what's going on in this issue overlaps with #66571, so I'm going to close this out in favor of that one.

TL;DR: It's not a simple matter to make `%Z` return an aware object without also specifying `%z`, and I'm not sure it would be a backward-compatible change even if we could. There are options in that thread for a future where it's possible to get an aware object with just `%Z`, though you'd need to pass in additional information (which obviates the backwards compatibility issue anyway).
